bpython/autocomplete.py

```diff
@@ -543,13 +546,31 @@ def get_completer(completers, cursor_offset, line, **kwargs):
         complete_magic_methods is a bool of whether we ought to complete
             double underscore methods like __len__ in method signatures
     """
-
     for completer in completers:
         matches = completer.matches(cursor_offset, line, **kwargs)
         if matches is not None:
-            return sorted(matches), (completer if matches else None)
+            if len(matches) > 0:
+                matches = sort_by_underscore(matches)
+            return matches, (completer if matches else None)
     return [], None
 
+def sort_by_underscore(matches):
+    """Sort single underscore attributes before double underscore ones.
+    """
+    matches = sorted(matches)
```

Comment:
Doesn't really matter, but I'd prefer `matches.sort()` instead of reassigning to the variables - I guess this keeps the matches list being passed in safe from modifications, but if we're doing this we should change the docstring from "sort asdfasdf" to "Returns a sorted list" because sort implies mutating the list imo.